master/buildbot/worker/kubernetes.py

```diff
@@ -0,0 +1,303 @@
+# This file is part of Buildbot. Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import copy
+from datetime import datetime
+from datetime import date
+import hashlib
+import socket
+
+from twisted.internet import defer
+from twisted.internet import threads
+from twisted.python import log
+
+from buildbot import config
+from buildbot.interfaces import IProperties
+from buildbot.interfaces import IRenderable
+from buildbot.interfaces import LatentWorkerFailedToSubstantiate
+from buildbot.process.properties import Interpolate
+from buildbot.process.properties import Properties
+from buildbot.util import unicode2bytes
+from buildbot.worker.docker import AbstractLatentWorker
+
+from future.utils import integer_types
+from future.utils import PY2
+from future.utils import string_types
+from future.utils import text_type
+
+from zope.interface import implementer
+
+try:
+    from kubernetes import config as kube_config
+    from kubernetes import client
+except ImportError as exc:
+    kube_config = None
+    client = None
+
+if PY2:
+    FileNotFoundError = IOError
+
+
+@implementer(IRenderable)
+class KubeRenderable(object):
+
+    untouched_types = integer_types + (
+        float, bool, bytes, datetime, date, type(None)
+    )
+
+    def __init__(self, kube_obj):
+        self.kube_obj = kube_obj
+
+    @defer.inlineCallbacks
+    def getRenderingFor(self, props):
+        res = yield self.recursive_render(
+            copy.deepcopy(self.kube_obj),
+            props
+        )
+        defer.returnValue(res)
+
+    @defer.inlineCallbacks
+    def recursive_render(self, obj, props):
```

Comment:
This would be so much simpler if we would just use the REST api.

We would just use json like objects that are already deeply renderable.